parlai/tasks/wizard_of_wikipedia/agents.py

```diff
@@ -392,6 +392,7 @@ def get(self, episode_idx, entry_idx=0):
         text = dialog_entry_1['text']
         labels = [dialog_entry_2['text']]
 
+        assert isinstance(self.add_topic, bool)
```

Comment:
do we need this assert?

Reply:
I think it'd be good to have it - I had a run train on the wrong data silently because self.add_topic was set to the string `"False"`, which the line below interpreted as `True` due to Python's sloppy polymorphic overloading of what `if` means

Reply:
how did that happen?

Reply:
Ha, it was a while ago, but I believe that that param was set to `"False"` as a result of the bug closed by https://github.com/facebookresearch/ParlAI/pull/2374 , where bool flags embedded within task names would be evaluated as strings. If it's really annoying to have this check we can remove it, but in general I like having checks like this that prevent silent errors, even if something else has to go wrong in order to trigger that error. Happy to discuss more :)

Reply:
I'm ok with leaving it, but yeah i suppose that's another reason that we should probably avoid flags in task names

Reply:
Agreed - it's a useful hack to be able to put flags in task names but it's definitely problematic